### DISCUSSION TITLE
Welcome to Pelican Discussions!

### DISCUSSION THREAD

**Discussion by justinmayer** [getpelican/pelican#3041] [Announcements]
## 👋 Welcome!
  Pelican Discussions is a place to connect with other members of our community. We hope that you:
  * first read _[How to Get Help](https://docs.getpelican.com/en/latest/contribute.html#how-to-get-help)_ and then ask questions you’re wondering about
  * share ideas
  * engage with other community members
  * are open-minded and welcome others, remembering that this is a community we
  build together 🌈 

  To kick off this new aspect of the Pelican community, comment below with a self-introduction and perhaps some links to your Pelican-powered sites. Excited to see what you all are building!

**Comment by silverhook** [comment]
Thanks, @justinmayer for this. I hope it serves the community well.

I’m a lawyer by trade and hacker by heart and have been blogging since the turn of the century/millennium and have been [using Pelican since 2013](https://matija.suklje.name/new-blog-pelican) – and I managed to migrate all but the very first posts to my current Pelican blog. I even wrote a [lengthy post](https://matija.suklje.name/5th-incarnation-of-hooks-humble-homepage) about how I managed to pull that through.

In the Pelican sphere, I’m part of the [Pelican Elegant team](https://github.com/Pelican-Elegant/elegant/), admittedly probably the least technical one and as such not super active there really.

My blog [Hook’s Humble Homepage](https://matija.suklje.name) is divided into four categories:
- _Anima_: Everything relating mainly to life – from deep thoughts to mundane moments.
- _Ars_: Everything relating to art – books, films, music, games &c.
- _Ius_: Everything relating to law – including copyright, patents, criminal law and all other bodies of law.
- _Tehne_:  Everything that is of a more technical nature – diverse HowTos, new exciting IT solutions as well as just server/blog software updates.

My coding skills are extremely limited, so I probably won’t be much help here either. If you need a tip or two about licensing though ;)

**Reply by lcfd** [reply]
Wow @silverhook, a good and well-kept bio. Nice to meet you.
I've seen the Elegant theme and team in the past, it is a good project!

No matter the technical qualities, any feedback or kind of help is always appreciated.

**Comment by ndeville** [comment]
Thanks @justinmayer & team for all your hard work on Pelican! I love it!

I discovered it 3 months ago today, and have migrated a dormant blog on Ghost with 50 posts to a new "notes" site on Pelican already with 430 notes in it: 

https://notes.nicolasdeville.com/

My goal was to do notecasting or note sharing, ie not polished posts but rather just my own notes, shared via a static site. 
Writing & managing all in VS Code which makes it a breeze. 
They are intended to help me, and perhaps others - but not aiming for audience building. 

Thanks to the implementation of Stork on top, search is lightning fast.

My (unpolished) note about getting started with Pelican: https://notes.nicolasdeville.com/python/library-pelican/

which continues in this note, where I continue to build out my site: https://notes.nicolasdeville.com/projects/notes/

I'm experienced in Python scripting but not in building software, so not sure if/how I can help and give back. 

Hope to see Pelican grow and gain even more popularity - it definitely deserves it!

**Reply by lcfd** [reply]
Nice to meet you @ndeville!

Your use-case is a thing that I thought also for myself. At the moment I'm using Obsidian to take notes that have to last and it probably will work amazingly with Pelican.

> so not sure if/how I can help and give back

You can always give back by writing some feedback, finding bugs and reporting them, and reading the documentation to report what's not clear and what is missing. Those are good starting points.

> it definitely deserves it!

I agree! I hope to see you around often!

**Reply by silverhook** [reply]
Nice one! I sometimes do the same on my blog too. Especially when I find a technical solution that I’m bound to forget, or when I compare different things like e.g. different DE razor blades or shaving oils (yeah, it’s an outlier on my blog, but still)

The great side-effect of writing such notes is that eventually people will find your notes when they search for an issue they have online. So such blogs are very valuable!

**Reply by ndeville** [reply]
@lcfd didn't Obsidian.. looks good. Sticking with Pelican though ;) 
I struggle to report bugs because I always think the most likely weak link is between the chair and the keyboard!
What's the best way to report feedback? Or report documentation update suggestions?

**Reply by ndeville** [reply]
@silverhook I agree.. I've been helped so many times by random people's static static sites, that I thought someone might be helped with mine. 

"lawyer by profession and hacker by heart" is quite unusual! :)
Looked for your post on DE razor blades but couldn't find it - search does not work?
I go for Wilkinson Sword Double Edge Blades for my Merkur 34C but I might be missing a better alternative ;)

**Reply by lcfd** [reply]
> Sticking with Pelican though ;)

Obsidian is just a text editor like vscode, you don't have to give up with Pelican.
It can be just a convenient way to link and manage your notes because is thought to manage markdown files.

> What's the best way to report feedback? Or report documentation update suggestions?

Here in Discussions will be perfect! In this place, you can just ask whatever you want like in a normal forum fashion.

> likely weak link is between the chair and the keyboard!

It took me a while to figure it out 🤭 Don't worry! 
Just follow the guidelines, there are no silly or limited questions. Maybe check first if it has already been asked.
However, the goal is to create a sort of integration to the documentation.

**Reply by silverhook** [reply]
> "lawyer by profession and hacker by heart" is quite unusual! :)

Hehe, yeah, there’s not many of us :)

> Looked for your post on DE razor blades but couldn't find it - search does not work? I go for Wilkinson Sword Double Edge Blades for my Merkur 34C but I might be missing a better alternative ;)

Yeah, search is currently borked on Pelican Elegant https://github.com/Pelican-Elegant/elegant/issues/789 – super welcome if someone can take a look, Talha is ultra-busy these days.

You’re probably looking for this link (and I definitely should update it, as many are missing):

http://matija.suklje.name/my-favourite-razor-blades-for-shaving-with-oil

(P.S. I recently received a bunch of _new_ Tiger blades, and have to say I like the new [Tiger Platinum](https://www.czechblades.cz/en/catalog/stainless-blades/tiger-platinum) even more than my daily driver vintage Tiger)

**Comment by justinmayer** [comment]
The best way to report feedback or documentation update suggestions would be to create a new discussion thread here. 😊

**Reply by ndeville** [reply]
Thanks for confirming! I'll do that moving forward.. :)

**Comment by minchinweb** [comment]
Hi I'm @MinchinWeb! I've been using Pelican for years, and continue to find it incredible. For maintaining my own site, having a static site has been wonderful because I can leave it alone for 6 months (or more) at a time, and *it doesn't break*! My personal site is a collection of Pelican "sites":

- my root site [minchin.ca](https://minchin.ca/): Previously, it was hand-crafted HTML, but in [2014](https://github.com/MinchinWeb/minchinweb.github.io/commit/3564b50cf462522a887a4dfb62daed23c1671f7b) I moved it to Pelican, likely in part in an effort to make updating the theme between this and my blog easier. This is a Pelican site that is "pages" only, and has no articles (i.e. blog content). The site is generated by GitHub Actions and hosted on GitHub Pages. Source is here: <https://github.com/MinchinWeb/minchinweb.github.io>
- my blog ([blog.minchin.ca](https://blog.minchin.ca/)): a "traditional" Pelican blog. Originally hosted on Blogger, but exported in [2016](https://github.com/MinchinWeb/blog.minchin.ca/commit/11534e8751724a910d2449a4c00c9352209a0180), likely as I wanted more control over the output (Blogger was doing weird things with mobile layouts at the time...). There are currently 257 posts. The theme is custom ([seafoam](https://github.com/MinchinWeb/seafoam)) and plugins are used extensively. The Pelican site was originally hosted on GitHub Pages, but is currently built and hosted on Netlify (at the time of the move to Netlify, GitHub pages didn't support HTTPS; it does now). Source is here: <https://github.com/MinchinWeb/blog.minchin.ca>
- genealogy sub-site at [minchin.ca/genealogy](http://minchin.ca/genealogy/): this is first generated from a GEDCOM to HTML converter, and then run through Pelican so that the style matches the rest of the site. The (sub-)site is more than 10,000 pages generated! Currently generated offline, and then uploaded to GitHub, and hosted from GitHub Pages (as a "project" page). Source is here: <https://github.com/MinchinWeb/genealogy>

I've also used Pelican a couple of times for company sites, and find it works well as long as you don't need interactive features (like a "contact us" page that feeds into a database directly). The page loading times are neigh unbeatable!

I have also used a private, local-only Pelican site that is a collection of my notes. I find generated my (plain text) notes into a Pelican site makes them much easier to read and review.

I've also collected, packaged, polished, and/or written a collection of plugins to make my sites do even more. My personal philosophy has been to use plugins available to be installed via `pip` to ensure consistent builds (very helpful when you step back in months later), and to package plugins not thus available and post them to PyPI myself. One of the most exciting developments (at least to me) in Pelican's history was when it gained the ability to autoload plugins (with v4.5)! "My" plugins include:

- [seafoam](https://github.com/MinchinWeb/seafoam) -- my website theme. Packaged as a plugin, it auto-loads and auto-configures itself!
- [summary](https://github.com/MinchinWeb/minchin.pelican.plugins.summary) -- `minchin.pelican.plugins.summary` -- "allows easy, variable length summaries directly embedded into the body of your articles."
- [cname](https://github.com/MinchinWeb/minchin.pelican.plugins.cname) -- `minchin.pelican.plugins.cname` -- "creates a CNAME file in the root of your output directory. This is useful when you are publishing your site to GitHub Page on a [custom domain](https://help.github.com/articles/using-a-custom-domain-with-github-pages/)."
- [static comments](https://github.com/MinchinWeb/minchin.pelican.plugins.static_comments) -- `minchin.pelican.plugins.static_comments` -- Yes, I have comments on my site! and without the need of another server! Comments are submitted via email. Honestly, it works surprisingly well for me. I'm not sure how well it would scale if you were getting a dozen (or more) comments a day as there are some manual steps involved.
- [optimize images](https://github.com/MinchinWeb/minchin.pelican.plugins.optimize_images) -- `minchin.pelican.plugins.optimize_images` -- "Applies lossless compression on JPEG and PNG images, with no effect on image quality."
- [autoloader](https://github.com/MinchinWeb/minchin.pelican.plugins.autoloader) -- `minchin.pelican.plugins.autoloader` -- I feel like this is a sort of "secret sauce"! It extends plugin autoloading to additional namespaces (beyond the default of `pelican.plugins`). My other plugins all rely on this.
- [post stats](https://github.com/MinchinWeb/minchin.pelican.plugins.post_stats) -- `minchin.pelican.plugins.post_stats` -- "calculates various statistics about a post and store them in an article.stats dictionary", things like reading time. Theme support is needed to actually see these.
- [image process](https://github.com/pelican-plugins/image-process) -- `pelican-image-process` -- "let you automate the processing of images based on their class attribute. Use this plugin to minimize the overall page weight and to save you a trip to Gimp or Photoshop each time you include an image in your post." Useful for things like cropping or resizing images. This is actually a plugin someone else wrote, I picked up when it seemed to be abandoned (and I needed updates for it to work for me), and eventually returned to community stewardship. It's probably the most popular plugin on this list.
- [no Jekyll](https://github.com/pelican-plugins/nojekyll) -- `minchin.pelican.plugins.nojekyll` -- "creates a .nojekyll file in the root of your output directory. This is useful when you are publishing your site to GitHub Pages as it keeps your site from being run through GitHub's default Jekyll site generator. This has a side effect of make your updated site go live faster."
- [Jinja Filters](https://github.com/pelican-plugins/jinja-filters) -- `pelican-jinja-filters` -- "provides a selection of functions (called filters) for templates to use when building your website. They are packaged for Pelican".

I also have a few in-progress plugins/projects for Pelican, that I hope will eventually see the light of day. These are all very early stage (like not even proof-of-concept yet):

- [exodus](https://github.com/MinchinWeb/exodus) -- to go from my desktop genealogy program to a static website. I expect I'll first use this to extract the data into a form that can be passed to Pelican, and then use Pelican as a page generation engine. Last time I was working on this, I was stuck trying to deal with a non-standard collation in the SQLite database the desktop program generates.
- [GPX Reader](https://github.com/MinchinWeb/gpx-reader) -- a reader for Pelican that would turn GPX files (generally, history tracks generated from a GPS receiver, or a program on your phone) into something pretty (maps, stats, maybe elevation and speed graphs) that can be understood by Pelican.
- [ePub Reader](https://github.com/MinchinWeb/minchin.pelican.readers.epub) -- a reader for Pelican for ePub (eBook) files. In theory, this shouldn't be too complicated as ePubs are (generally) a collection of HTML files (plus some extras) under the hood. This would re-display them as part of a Pelican site. I'd like to extend this further by somehow backing my reading notes into the generated site; file format and CSS styling to be determined.
- [Book page numbering](https://github.com/MinchinWeb/blog.minchin.ca/issues/103) -- (so early stage, it only exists as a filed issue on my blog!) -- the idea is to assign page numbers and page breaks throughout the generated Pelican site, to allow you to learn into a book metaphor. Maybe it's a dumb idea, but I think it could be an interesting touch.

Hopefully, some of my work has been helpful to you to! I hope to be able to use Pelican for many years to come.

**Comment by sarnold** [comment]
Howdy!  I originally started with pelican to migrate my (tiny) office site to something we could all contribute to (where "we" is generally less than a half-dozen engineers plus a couple of part-time business-y helpers) but then I got busy and stopped writing random content pages. So now I'm trying to "force" myself to publish more often by making a personal pelican site for Github pages.

I got the basics built with an example today but I could not find the "chat" help I saw mentioned in a couple of (closed) issues on github. Questions:

1. Does the IRC help channel still exist?
2. Is this the preferred place for questions *before* making in Issue on giithub?

I went looking for that ^^ because of initial build errors when I added the custom 404.md file and looked at the github link which told me to add the "front matter" block with `permalink: /404.html` which did *not* work with the example 404 markdown.  I did get it working when included the title bits *in the front matter* so now it looks like this:

```
---
title: Not Found
status: hidden
save_as: 404.html

permalink: /404.html
---

The requested item could not be located. Perhaps you might want to check
the [Archives](/archives.html)?
```
Is this actually the correct way to do it?  Thanks for any feedback, as well as for the slick web tooling in Python (one of my top 2 favorite languages!).

**Reply by sarnold** [reply]
Oh, I'm mostly doing contract engineering these days, with as much FOSS as possible. Besides that I have a lot of other Github repos with various development-related tools along with some template repos for python/cmake/DID-based formal docs.

* work stuff: https://github.com/VCTLabs
* my stuff: https://github.com/sarnold

**Reply by minchinweb** [reply]
Here's how I did it:  https://github.com/minchinweb/strathcona-power-website/blob/pelican/content/pages/404.md  (click "Raw" to see the front matter)

I used `slug: 404` rather than `permalink`...

**Reply by sarnold** [reply]
Thanks!
